CHANGELOG.md

```diff
@@ -3,6 +3,10 @@ Note: This is in reverse chronological order, so newer entries are added to the
 Swift Next
 -----------
 
+* [#5966]
+
+  Plugin compilation can be influenced by using `-Xhost-swiftc` arguments in the SwiftPM command line. This is similar to the existing mechanism for influencing the manifest compilation using `-Xmanifest` arguments. Manifest compilation will also be influenced by `-Xhost-swiftc`, but only if no other `-Xmanifest` arguments are provided. Using `-Xmanifest` will show a deprecation message. `-Xmanifest` will be removed in the future.
```

Comment:
I don't particularly like `-Xhost` prefix. We recently had a discussion what exactly "host" means, especially in a cross-compilation context, and there we opted in for "build-time" instead of "host" nomenclature. I realize though that "build-time" doesn't make sense here, and I don't have a good alternative to propose instead. But I don't think that `-Xhost` is obvious and self-descriptive enough, in my opinion.

Reply:
As I said in the comments above, if someone proposes a better spelling, I am all game. I use "host" because it is already used in the code (example: https://github.com/apple/swift-package-manager/blob/14d05ccaa13b768449cd405fff81d630a520e04a/Sources/PackageModel/Destination.swift#L159).

Reply:
What's your opinion on `-Xbuild-tools-swiftc`? The original proposal for plugins was named "Extensible Build Tools", and a package manifest is a build tool too, from a certain perspective.

Reply:
seems fine to me. @abertelrud @neonichu @drodriguez ?

Reply:
IMO too narrow, e.g. a command plugin is not a build tool, but I don't have a good alternative to propose

Reply:
@drodriguez apologies about the back and forth on the naming and you patience with this.

@MaxDesiatov is `build-tool` so it is consistent with the naming scheme for CC? or something different?

Reply:
I'm referring to the original "Extensible Build Tools" proposal for SwiftPM plugins. Overlap with cross-compilation nomenclature is accidental, although I do like that it becomes somewhat similar.

Reply:
IMO, the proposal name should also have changed once we introduced command plugins.